### PR TITLE
Added tests using RSpec

### PR DESCRIPTION
**Pull request by sebalp100**
# Added tests using RSpec

For this milestone, we added unit tests using RSpec for the following classes:

- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- Decorator

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    end
+
+    context 'when person is underage and does not have parent permission' do
+      let(:person) { Person.new(16, 'David', parent_permission: false) }
+
+      it 'returns false' do
+        expect(person.can_use_services?).to eq(false)
+      end
+    end
+  end
```
Comment (danmainah) [inline spec/person_spec.rb:53]: - [x] @sebalp100 and @enoqueJonas great work on your tests however you forgot to ad tests for correct_name and of_age methods

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      @classroom.students << @student
+      expect { @student.label = 'C+' }.not_to(change { @classroom.students.size })
+    end
+  end
+
+  describe '#play_hooky' do
+    it 'prints a message' do
+      expect { @student.play_hooky }.to output("¯(ツ)/¯\n").to_stdout
+    end
+  end
```
Comment (danmainah) [inline spec/student_spec.rb:39]: - [x] @team great work on your tests however you forgot to add tests for initialize method, kindly add tests for this method.

File: spec/decorator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  describe '#correct_name' do
+    it 'returns the correct name' do
+      expect(@decorator.correct_name).to eq('John Smith')
+    end
+  end
+end
+
+describe CapitalizeDecorator do
```
Comment (danmainah) [inline spec/decorator_spec.rb:16]: - [ ] [optional] It would be nice if each class had its own tests file

Comment (danmainah) [reply spec/decorator_spec.rb:16]: - [x] @team you forgot to add tests for the nameable class, kindly add the tests

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @sebalp100 and @enoqueJonas , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/person_spec.rb (hunk 1 above)
Comment (sebalp100) [reply spec/person_spec.rb:53]: @danmainah Hi, the of_age method is private so we are indirectly testing it with the can_use_services method

**Comment by sebalp100** [comment]
Changes required implemented

**Review by alexander16108 — APPROVED** [review]
Hi @sebalp100  @enoqueJonas , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**